src/message_router/domain_types.rs

```diff
@@ -1025,7 +1017,10 @@ impl FipaMessage {
     pub fn try_new_validated(
```

Comment:
This function is named poorly. Why isn't it just `try_new`? Also, why an extra layer of abstraction with `FipaMessageParams`? Why not just take the params as normal function arguments?